src/DynamoCore/Graph/Nodes/NodeInputData.cs

```diff
@@ -92,7 +92,7 @@ public class NodeInputData
         /// The index of the selected item.
         /// </summary>
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore)]
-        public int SelectedIndex { get; set; }
+        public int? SelectedIndex { get; set; }
```

Comment:
it's likely this is an API break.
I don't know if we care - but I think it fits in


❌ DISALLOWED: Changing the type of a member

The return value of a method or the type of a property or field cannot be modified. For example, the signature of a method that returns an Object cannot be changed to return a String, or vice versa.

this may cause binary compatibility issues with any code that referenced this member, can you check?

Reply:
I think you are right.  That was my reading of it as well.  I do think we should probably make the change though.  I will double check.

Reply:
Hey - finally time to do this?

Reply:
Marked as 3.0

Reply:
Yep let's do this.  Just updated the branch @mjkkirschner @QilongTang